### ISSUE TITLE
no 4g device connected for gui> 9.3.71

### ISSUE THREAD

**Issue by kisspachyousee** [Ansuel/tch-nginx-gui#1043]
Device Model/Firmware: tg789vac v2
GUI Version: 9.3.71

from the version after 9.3.71, including the latest stable version, the internet key will no longer be recognized. I use the k5160 key.
seen the problem, I started from version 9.1.2 and I made progressive upgrades. when I got to version 9.3.77 (I skipped version 9.3.75) it stopped working.
once version 9.3.77 is installed, I cannot downgrade to a version where the key worked, because it will still not work, so a reset from the key is required. in addition, once the reset is done, directly putting the version that worked 9.3.71 involves the malfunction of the interface with error 500. so I put a version where the gui worked and then I updated to 9.3.71 to have the connection with key.

dalla verisione successiva alla 9.3.71, compresa l'ultima versione stabile, non viene riconosciutà più la chiavetta internet. io uso la chiavetta k5160. 
visto il problema, sono partito dalla versione 9.1.2 e ho fatto upgrade progressivi. quando sono arrivato alla versione 9.3.77 (ho saltato la versione 9.3.75) ha smesso di funzionare.
una volta installata la versione 9.3.77, non posso fare un downgrade ad una versione dove la chiavetta funzionava, perchè continuerà a non funzionare, quindi è necessario un reset dal tasto. in più, una volta fatto il reset, mettere direttamente la versione che funzionava 9.3.71 comporta il mal funzionamento dell'interfaccia con errore 500. quindi ho messo una versione dove la gui funzionava e poi ho aggiornato alla 9.3.71 per avere la connessione con chiavetta.

**Comment by FrancYescO** [comment]
what firmware? have you tried to put directly last stable after a reset?

**Comment by kisspachyousee** [comment]
firmware is version 17.2, it should be Uno.
If I put the last gui directly, it happens to me that the 4g doesn't work. says 0 connected devices.

**Comment by FrancYescO** [comment]
please post a debug zip from the last dev
try if issuing `/etc/init.d/mobiled restart` gives errors and try after `cp /rom/etc/init.d/mobiled /etc/init.d/mobiled`

**Comment by FrancYescO** [comment]
have you done a full reset or just a GUI reset?

**Comment by kisspachyousee** [comment]
No, Just gui reset. I Will try with a full reset.

**Comment by FrancYescO** [comment]
try also `cp /rom/usr/lib/lua/mobiled/scripthelpers.lua /usr/lib/lua/mobiled/scripthelpers.lua`
but we should find a way to fix it after a full reset (preserving only root) and a direct install of a version that is at least 9.6.65, otherwise debugging is getting harder.

**Comment by kisspachyousee** [comment]
with full reset same result. Gui 9.6.65 stable:
`root@dsldevice:~# cp /rom/usr/lib/lua/mobiled/scripthelpers.lua /usr/lib/lua/mob
iled/scripthelpers.lua
cp: '/rom/usr/lib/lua/mobiled/scripthelpers.lua' and '/usr/lib/lua/mobiled/scripthelpers.lua' are the same file
root@dsldevice:~# /etc/init.d/mobiled restart
Delaying Mobiled startup. Waiting for 4865 to terminate
sh: can't kill pid 4865: No such process
root@dsldevice:~# cp /rom/etc/init.d/mobiled /etc/init.d/mobiled
cp: '/rom/etc/init.d/mobiled' and '/etc/init.d/mobiled' are the same file
root@dsldevice:~# cp /rom/usr/lib/lua/mobiled/scripthelpers.lua /usr/lib/lua/mob
iled/scripthelpers.lua
cp: '/rom/usr/lib/lua/mobiled/scripthelpers.lua' and '/usr/lib/lua/mobiled/scripthelpers.lua' are the same file
`
this is dmesg log:
`
Tue Dec 22 12:45:48 2020 daemon.notice [11387]: [mobiled] (PlatformConfigure) runs PlatformConfigure-Main.check(timeout, 1)
Tue Dec 22 12:45:48 2020 daemon.notice [11387]: [mobiled] (WaitingForDevice) runs WaitingForDevice-Main.check(timeout, 1)
Tue Dec 22 12:45:48 2020 daemon.info [11387]: [mobiled] Executing "ubus send mobiled '{"dev_desc":"1-2","event":"device_added","dev_idx":1}'"
Tue Dec 22 12:45:48 2020 daemon.notice [11387]: [mobiled] (DeviceInit) runs DeviceInit-EntryExit.entry(1)
Tue Dec 22 12:45:48 2020 daemon.notice [11387]: [mobiled] DeviceInit-> Entry Function
Tue Dec 22 12:45:48 2020 daemon.err [11387]: [mobiled] Failed to initialize control channel
Tue Dec 22 12:45:48 2020 daemon.info [11387]: [mobiled] Executing "ubus send mobiled '{"dev_desc":"1-2","pdp_type":"ipv4v6","session_id":0,"session_state":"disconnected","dev_idx":1,"event":"session_state_changed"}'"
Tue Dec 22 12:45:48 2020 daemon.info [11387]: [mobiled] Destroy device 1
Tue Dec 22 12:45:48 2020 daemon.info [11387]: [mobiled] Removed statemachine 1
Tue Dec 22 12:45:48 2020 daemon.info [11387]: [mobiled] Executing "ubus send mobiled '{"dev_desc":"1-2","event":"device_removed","dev_idx":1}'"
Tue Dec 22 12:45:48 2020 daemon.info [11387]: [mobiled] Adding statemachine 1
Tue Dec 22 12:45:49 2020 daemon.err [11387]: [mobiled] Received event "session_state_changed" for unknown device


`
[DebugHelper2020-12-22-1136(gui 9.6.65 dopo reset).tar.gz](https://github.com/Ansuel/tch-nginx-gui/files/5730943/DebugHelper2020-12-22-1136.gui.9.6.65.dopo.reset.tar.gz)

**Comment by FrancYescO** [comment]
try executing `mobiled` from shell

**Comment by kisspachyousee** [comment]
ok, but I have to try with the new dev release or with mobiled via shell?
I also noticed that the rndis_host devices are not recognized, while in the DGA4132 model they are recognized

**Comment by FrancYescO** [comment]
last dev should fix the issue please confirm..

rndis_host??

**Comment by kisspachyousee** [comment]
I tried to update from the interface and it gave me a corrupt file. I try to update from shell.
regarding rndis, sometimes I connect the smartphone to the usb port and enable usb tethering which generates a wan connection with the modem as described in the OpenWrt guide https://openwrt.org/docs/guide-user/network/wan/wwan / ethernetoverusb_rndis

**Comment by FrancYescO** [comment]
pretty sure it's due to different versions of mobiled, probably you are using a firmware with an updated version on 4132

try moving this this dir to the 789 https://github.com/FrancYescO/tch_firmware_extracted/tree/AGTHP_2.2.2_001_CLOSED/usr/lib/lua/libat

**Comment by kisspachyousee** [comment]
![image](https://user-images.githubusercontent.com/49493719/102941071-13ae7580-44b2-11eb-989c-fb10b4faf02d.png)
I updated from shell without resetting and it worked right away!

**Comment by kisspachyousee** [comment]
for the rndis I used the files you recommended but nothing to do. from the log it recognizes it but it seems that it does not distribute the connection in the lan, the same thing happened to me with the 4g connection and the voip that did not connect. I had solved the latter with a workaroud in the network settings.
in the DG4132 instead everything works without any modification.
I'll try to make some changes.

**Comment by FrancYescO** [comment]
what is the workaround used for rndis? probably we are looking at different issues, it turn the interface on but set no routing? can you send a debug log with both rndis connected from 4132 and from 789?

**Comment by kisspachyousee** [comment]
no, sorry maybe I explained myself wrong. with the tg789 the voip did not work with the 4g (while with the dg4132 yes). 
To fix this I edited the files in /etc/config as I did here [.](https://www.ilpuntotecnico.com/forum/index.php/topic,77519.msg255219.html#msg255219)
Now I tried to connect with rndis instead, which is recognized. from the log I see: `usb 1-1: new high-speed USB device number 3 using ehci-platform
[3039.710000] rndis_host 1-1: 1.0 usb0: register 'rndis_host' at usb-ehci-platform-1, RNDIS device, 02: 68: 02: 73: 12: 7a`
but i have no connection.
it is likely that I have to do the same procedure I did in the forum but this time choose usb0 instead of wwan0 to be able to have internet.
it seems to be another type of problem, unrelated to the previous one.

**Comment by FrancYescO** [comment]
Probably better to continue on forum or in a new issue the RNDIS question (fwics attaching my phone in USB tethering to a DGA4130 with 18.3 firmware seems to not add any new interface) just point out here in a new issue fixes that the GUI can directly apply

actually is there anything that requires you to use an older version of the GUI?

**Comment by kisspachyousee** [comment]
ok, i will open a discussion on the forum.
no, now I use the latest version of the gui and it seems to work fine!
